src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,334 @@
+			printOperatorInfoForBindJoinWithVALUES( (PhysicalOpBindJoinWithVALUES) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpFilter ) {
+			printOperatorInfoForOpFILTER( (PhysicalOpFilter) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpGlobalToLocal ) {
+			printOperatorInfoForOpGlobalToLocal( (PhysicalOpGlobalToLocal) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpHashJoin ) {
+			printOperatorInfoForOpHashJoin( (PhysicalOpHashJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpHashRJoin ) {
+			printOperatorInfoForOpHashRJoin( (PhysicalOpHashRJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpIndexNestedLoopsJoin ) {
+			printOperatorInfoForOpIndexNestedLoopsJoin( (PhysicalOpIndexNestedLoopsJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpLocalToGlobal ) {
+			printOperatorInfoForOpLocalToGlobal( (PhysicalOpLocalToGlobal) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpMultiwayUnion ) {
+			printOperatorInfoForMultiwayUnion( (PhysicalOpMultiwayUnion) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpNaiveNestedLoopsJoin ) {
+			printOperatorInfoForOpNaiveNestedLoopsJoin( (PhysicalOpNaiveNestedLoopsJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpParallelMultiLeftJoin ) {
+			printOperatorInfoForOpParallelMultiLeftJoin( (PhysicalOpParallelMultiLeftJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpRequest ) {
+			printOperatorInfoForRequest( (PhysicalOpRequest) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else if ( rootOp instanceof PhysicalOpSymmetricHashJoin ) {
+			printOperatorInfoForOpSymmetricHashJoin( (PhysicalOpSymmetricHashJoin) rootOp, out, indentLevelString, indentLevelStringForOpDetail );
+		}
+		else {
+			throw new IllegalArgumentException( "Unexpected operator type: " + rootOp.getClass().getName() );
+		}
+		for ( int i = 0; i < plan.numberOfSubPlans(); ++i ) {
+			planWalk( plan.getSubPlan(i), i, planLevel+1, plan.numberOfSubPlans(), out, indentLevelString );
+		}
+	}
+	
+	protected void printOperatorInfoForBinaryUnion ( final PhysicalOpBinaryUnion op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "binaryUnion (" + op.getID() + ") " );
+		printLogicalOperator( nameOfLogicalOp(op.getLogicalOperator()), op.getLogicalOperator().getID(), out, indentLevelStringForOpDetail );
```

Comment:
I would change the signature of `printLogicalOperator` such that you can pass the logical operator and instead of passing both the short name and the ID of it. Then, `printLogicalOperator` takes care of obtaining the short name and the ID. This would reduce the amount of source code here in all of these `printOperatorInfoFor...` functions, which should make the code a bit more readable.